From d538f29b987ff266dbc1e9769ceb4560b2ac020f Mon Sep 17 00:00:00 2001
From: Simone Weiß <simone.weiss@linutronix.de> 
Date: Fri, 14 Oct 2022 00:22:06 +0200
Subject: [PATCH] trust machine keyring (MoK) by default
Forwarded: not-needed

Debian always trusted keys in MoK by default. Upstream made it
conditional on a new EFI variable being set.
To keep backward compatibility skip this check and remove now
unused function that was used for this check only, also just return
true in trust_moklist directly.

This is an adaption of a patch first submited by Luca Boccassi
with `c3e0306959e6`.

Signed-off-by: Simone Weiß <simone.weiss@linutronix.de>
---
 .../platform_certs/machine_keyring.c          | 23 +------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

Index: linux-6.12.69/security/integrity/platform_certs/machine_keyring.c
===================================================================
--- linux-6.12.69.orig/security/integrity/platform_certs/machine_keyring.c
+++ linux-6.12.69/security/integrity/platform_certs/machine_keyring.c
@@ -43,38 +43,9 @@ void __init add_to_machine_keyring(const
 		pr_info("Error adding keys to machine keyring %s\n", source);
 }
 
-/*
- * Try to load the MokListTrustedRT MOK variable to see if we should trust
- * the MOK keys within the kernel. It is not an error if this variable
- * does not exist.  If it does not exist, MOK keys should not be trusted
- * within the machine keyring.
- */
-static __init bool uefi_check_trust_mok_keys(void)
-{
-	struct efi_mokvar_table_entry *mokvar_entry;
-
-	mokvar_entry = efi_mokvar_entry_find("MokListTrustedRT");
-
-	if (mokvar_entry)
-		return true;
-
-	return false;
-}
-
 static bool __init trust_moklist(void)
 {
-	static bool initialized;
-	static bool trust_mok;
-
-	if (!initialized) {
-		initialized = true;
-		trust_mok = false;
-
-		if (uefi_check_trust_mok_keys())
-			trust_mok = true;
-	}
-
-	return trust_mok;
+	return true;
 }
 
 /*
